### PR TITLE
Update Test Failure Issue Template

### PR DESCRIPTION
**Pull request by TanayParikh**
Was [using the test failure template](https://github.com/dotnet/aspnetcore/issues/33851) and thought the following changes may make filing an issue easier.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/50_test_failure.md [hunk 1]
```diff
@@ -7,18 +7,34 @@ assignees: ''
...
+
+-
+
 ## Error Message
 
 <!--
 Provide the error message associated with the test failure, if applicable.
 -->
 
+```
```
Comment (dougbu) [inline .github/ISSUE_TEMPLATE/50_test_failure.md:24]: nit:
```suggestion
``` text
```

File: .github/ISSUE_TEMPLATE/50_test_failure.md [hunk 2]
```diff
@@ -7,18 +7,34 @@ assignees: ''
...
+
+```
+
 ## Stacktrace
 
 <!--
 Provide the stack trace associated with the test failure, if applicable.
 -->
 
+```
```
Comment (dougbu) [inline .github/ISSUE_TEMPLATE/50_test_failure.md:34]: nit
```suggestion
``` text
```

**Review by dougbu — COMMENTED** [review]
I could go either way on this. On one hand, we do get too many bug reports in general that lack code blocks and are therefore extra-difficult to read. On the other, _removing_ the code blocks when either **Error Message** or **Stacktrace** aren't applicable becomes a burden.

In any case, is the problem just an understanding gap across our team❔ Would be surprised if many others use this template.

**Comment by BrennanConroy** [comment]
While we're changing this template, could we include a logs section? Most people leave out the logs when filing test failure issues and then in 2 weeks when the build is gone we can't see the logs anymore.

Also, the template should be changed to use `<details>`, it's a lot cleaner than listing out huge stack traces etc.
<details>
<summary>This is a title</summary>
Some logs here
</details>

**Comment by TanayParikh** [comment]
> On the other, _removing_ the code blocks when either **Error Message** or **Stacktrace** aren't applicable becomes a burden.

In most cases I expect it'll get filled out, and if it's left empty because it doesn't apply, it's not really a big deal in my opinion.

> In any case, is the problem just an understanding gap across our team❔ Would be surprised if many others use this template.

I assume this is almost exclusively used by the team. Purpose of this PR is just to make the motions of filing a test failure just a little bit easier rather than patching an understanding gap. 

> While we're changing this template, could we include a logs section? Most people leave out the logs when filing test failure issues and then in 2 weeks when the build is gone we can't see the logs anymore.
> Also, the template should be changed to use <details>, it's a lot cleaner than listing out huge stack traces etc.

Done and done.

**Review by dougbu — APPROVED** [review]
:ok:

**Pull request merged**